### ISSUE TITLE
Unexpected order when adding multiple elements to array

### ISSUE THREAD

**Issue by ryenus** [mikefarah/yq#2143]
**Describe the bug**

When trying to sequentially append two objects to an array, say `array + o1 + o2`, instead of appending o1 then appending o2, yq combined the objects o1 and o2, then appended the new object to the array.

```sh
yq -n '[] + {"a":1} + {"b":2}'
```

Version of yq: 4.X.X
Operating system: mac
Installed via: homebrew

**Input Yaml**
N/A

### **Command**
The command you ran:
```sh
yq -n '[] + {"a":1} + {"b":2}'
```

### **Actual behavior**

```yaml
- a: 1
  b: 2
```

### **Expected behavior**

```yaml
- a: 1
- b: 2
```

**Additional context**

**Comment by jandubois** [comment]
It looks like operators have right associativity, i.e. `a + b + c)` means `a + (b + c)`.

I'm not sure if this was intentional; I suspect it is a side-effect of transforming expressions into a postfix representation internally.

This is even more surprising for non-associative operators, like subtraction, if they are normally left-associative:

```console
$ yq -n '6 - 5 - 4'
5
```

Here the expression is processed as `6 - (5 - 4)`.

And of course due to the type-coercion properties of `+` even that operator is not associative in `yq`, as you have found out.

I don't see how this could be fixed without breaking backwards compatibility, but it would help a lot if this was documented (see #1648).

**Comment by ryenus** [comment]
Thank you @jandubois ! Verifying with subtraction indeed makes it more obvious:

| Command/Expression  | Result | Treated As 
|:-------------------:|:------:|---
| `jq -n '3 - 2 - 1'` | `0`    | `jq -n '(3 - 2) - 1'`
| `yq -n '3 - 2 - 1'` | `2`    | `yq -n '3 - (2 - 1)'` 😂

**Comment by jandubois** [comment]
@ryenus Just in case it wasn't obvious, you can avoid these issues by always adding parenthesis explicitly:

```console
$ yq -n '([] + {"a":1}) + {"b":2}'
- a: 1
- b: 2
```

**Comment by ryenus** [comment]
> @ryenus Just in case it wasn't obvious, you can avoid these issues by always adding parenthesis explicitly:

Agreed, for array concatenation, another workaround is to wrap the 2nd operand also in an array:

```
$ yq -n '[] + [{"a":1}] + {"b":2}'
- a: 1
- b: 2
```
